core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -110,17 +117,12 @@ private[spark] class EventLoggingListener(
         hadoopDataStream = Some(fileSystem.create(path))
         hadoopDataStream.get
       }
-
-    val compressionCodec =
-      if (shouldCompress) {
-        Some(CompressionCodec.createCodec(sparkConf))
-      } else {
-        None
-      }
+    val cstream = compressionCodec.map(_.compressedOutputStream(dstream)).getOrElse(dstream)
```

Comment:
I know this is not part of this change, but if feels like this whole method needs a try..finally to avoid leaking an open `dstream` when some exception is thrown.